ltsp/server/dnsmasq/ltsp-dnsmasq.conf

```diff
@@ -47,16 +47,16 @@ dhcp-mac=set:rpi,dc:a6:32:*:*:*
 # PXE services in non proxy subnets sometimes break UEFI netboot, so tag:proxy.
 pxe-service=tag:proxy,tag:!iPXE,X86PC,"undionly.kpxe",ltsp/undionly.kpxe
 pxe-service=tag:proxy,tag:!iPXE,X86-64_EFI,"snponly.efi",ltsp/snponly.efi
-pxe-service=tag:proxy,tag:iPXE,X86PC,"ltsp.ipxe",ltsp/ltsp.ipxe
-pxe-service=tag:proxy,tag:iPXE,X86-64_EFI,"ltsp.ipxe",ltsp/ltsp.ipxe
+pxe-service=tag:proxy,tag:iPXE,X86PC,"ltsp.ipxe","http://${proxydhcp/dhcp-server}/ltsp/ltsp.ipxe"
+pxe-service=tag:proxy,tag:iPXE,X86-64_EFI,"ltsp.ipxe","http://${proxydhcp/dhcp-server}/ltsp/ltsp.ipxe"
```

Comment:
These NICs: `00:02:c9:e9:62:c0,  06:00.0 Ethernet controller [0200]: Mellanox Technologies MT27500 Family [ConnectX-3] [15b3:1003]`

...have an internal iPXE implementation that might be older, but in any case it evaluates `http://${proxydhcp/dhcp-server}/ltsp/ltsp.ipxe` to `http:///ltsp/ltsp.ipxe`, so the clients fail to boot.
If I hard code an IP there, it works.

I think we need to check which iPXE version started supporting that, and send undionly.kpxe if it's an earlier version.

Reply:
Fully agree with you, I have exactly same cards, and didn't solved this yet.

Reply:
@alkisg, how can we check the version, is there any list of service variables used by ipxe? I haven't found any.

Reply:
Check these lines in dnsmasq.conf:

```shell
# "39" means "recent iPXE with menu support": http://ipxe.org/howto/dhcpd
dhcp-match=set:iPXE,175,39
```

Those are pretty much all I know. Currently I'm trying to load undionly.kpxe to any client that supports 175,38, which is VLAN, as those problematic NICs support VLAN and the default ipxe doesn't.

Reply:
Hi, I made some investigation:
* Here is list of predefined variables for iPXE: https://ipxe.org/cfg
* These Mellanox cards are using own [Mellanox FlexBoot firmware](https://ru.mellanox.com/page/products_dyn?product_family=34), it is based on ipxe version with same tag: `1.0.0+ (8990d)`, so we can't use `${version}` for check available features for it.